rfd/0058-package-distribution.md

```diff
@@ -0,0 +1,75 @@
+| YUM channel support via URI                        | No (one URI only)          | Yes                                                                                                          | Yes                                                                     | Yes                                            | Yes                                                                               |                                                                       |
+| Channel support notes                      | No support currently       | Can do anything with some reconfiguration                                                                    | Can do anything we care about with some reconfiguration                 | Very limited, no good solution                 | Missing some core features, would require CDN to rewrite HTTP header for requests |
+| Monitoring and alerting support            | In house, poll based only  | In house, poll based only                                                                                    | Webhooks on important events                                            | In house, poll based only                      | In house, poll based only (only supported in AR for Docker images)                |
+| Supports self hosting                      | N/A                        | N/A                                                                                                          | Yes                                                                     | Yes                                            | No                                                                                |
+| Has official Terraform provider            | N/A                        | N/A                                                                                                          | Yes                                                                     | No                                             | Yes                                                                               |
+| Pricing ($)                                | N/A                        | N/A                                                                                                          | $700/month                                                              | $700/month                                     | $0.1/GB/month stored, $0.09/GB/month egress                                       |
+| Notes:                                     |                            | Can do anything we want, just depends on the amount of initial and reoccuring engineering effort is required | Higher complexity, but supports pretty much any use case we'd ever need | Easy to use but probably not the best solution | Still in preview, not generally available                                         |
+
+### Signing key management
```

Comment:
Could you expand on the signing key management a bit:
- How/where is it going to be stored?
- What is the process of rotating it?

Reply:
To answer your questions (I can add this to the RFD if you'd like):
* How/where is it going to be stored? - I don't really have a good answer for this other than it won't change from how it's currently setup. Right now Drone has access to it, but I don't know if it's stored in Drone or elsewhere and I don't think I have access to Drone instance config to determine this.
* What is the process of rotating it? - There is no good way to do this with any solution with how APT is designed. Changing the key will, in call cases, cause a breaking change for our customers. Should the key become compromised we would need to inform our customers, create a new key, and resign the repositories.

Reply:
Drone secrets for the teleport repository are stored, encrypted, in [this file](https://github.com/gravitational/ops/blob/master/drone/secrets/teleport/secrets-teleport.yaml) in the [ops](https://github.com/gravitational/ops/) repository. They are loaded into drone as [drone secrets](https://docs.drone.io/secret/) and referenced in the `.drone.yml` file in the root of the teleport repository. An example of the signing key used is here: https://github.com/gravitational/teleport/blob/8b00efedaa35914752c21ad093dffc451fd04629/.drone.yml#L4929